src/implementations/config/create.ts

```diff
@@ -0,0 +1,192 @@
+            return await getDocumentStoreAddress(encryptedWalletPath);
+
+          case formType === "VERIFIABLE_DOCUMENT" && identityProofType === "DNS-DID":
+            info(`Creating temporary DNS for DID`);
+            return (
+              (await createTemporaryDns({
+                networkId: 3,
+                publicKey: `did:ethr:0x${walletObject.address}#controller`,
+                sandboxEndpoint: sandboxEndpointUrl,
+              })) || ""
+            );
+
+          case formType === "TRANSFERABLE_RECORD":
+            return await getTokenRegistryAddress(encryptedWalletPath);
+
+          default:
+            throw new Error("Invalid form detected in config file, please update the form before proceeding.");
+        }
+      } else {
+        throw new Error("Invalid form detected in config file, please update the form before proceeding.");
+      }
+    };
+
+    // loop through the form template to check the type of forms
+    const typesOfForms: TypesOfForms[] = forms.map((form) => {
+      const identityProofTypes = form.defaults.issuers.map((issuer: Issuer) => issuer.identityProof?.type);
+      return {
+        type: form.type,
+        identityProofTypes: identityProofTypes,
+      };
+    });
+
+    // generate doc store, token registry and DNS based on the form type in the form template
+    const documentStoreAddress = await getContractAddress(typesOfForms, "VERIFIABLE_DOCUMENT", "DNS-TXT");
+    const verifiableDocumentDnsTxtName = documentStoreAddress
+      ? await createTemporaryDns({ networkId: 3, address: documentStoreAddress, sandboxEndpoint: sandboxEndpointUrl })
+      : "";
+    const verifiableDocumentDnsDidName = await getContractAddress(typesOfForms, "VERIFIABLE_DOCUMENT", "DNS-DID");
+    const tokenRegistryAddress = await getContractAddress(typesOfForms, "TRANSFERABLE_RECORD", "DNS-TXT");
+    const tokenRegistryDnsName = tokenRegistryAddress
+      ? await createTemporaryDns({ networkId: 3, address: tokenRegistryAddress, sandboxEndpoint: sandboxEndpointUrl })
+      : "";
+
+    const updatedForms = updateForms(
+      forms,
+      documentStoreAddress,
+      tokenRegistryAddress,
+      walletObject,
+      verifiableDocumentDnsTxtName || "",
+      verifiableDocumentDnsDidName,
+      tokenRegistryDnsName || ""
+    );
+
+    const updatedConfigFile = updateConfigFile(configFile, wallet, updatedForms);
+
+    const configFileName = "config.json";
+    const outputPath = path.join(outputDir, configFileName);
+    fs.writeFileSync(outputPath, JSON.stringify(updatedConfigFile, null, 2));
+    return outputPath;
+  } catch (e) {
+    if (e instanceof Error) {
+      throw new Error(e.message);
+    }
+  }
+};
+
+const getConfigFile = async (configType: string, configTemplatePath: string): Promise<ConfigFile> => {
+  switch (configType) {
+    case "tradetrust":
+      return tradetrustConfig as ConfigFile;
+
+    default:
+      return JSON.parse(await readFile(configTemplatePath));
+  }
```

Comment:
why switch statement though?

Reply:
just in case in future we need to add more case, then it will be easier to add?